src/resdk/collection_tables.py

```diff
@@ -242,15 +273,133 @@ def _cache_file(self, data_type: str) -> str:
+            descriptors.append(sample.descriptor)
+
+        df = pd.json_normalize(descriptors).set_index("sample_name")
+
+        # Keep only numeric / string types:
+        column_types = {}
+        prefix = "XXX"
+        for (schema, _, path) in iterate_schema(
+            sample.descriptor, sample.descriptor_schema.schema, path=prefix
+        ):
+            field_type = schema["type"]
+            field_name = path[len(prefix) + 1 :]
+
+            # This can happen if this filed has None value in all descriptors
+            if field_name not in df:
+                continue
+
+            if field_type in ["basic:string:"]:
+                column_types[field_name] = str
+            elif field_type in ["basic:integer:"]:
+                column_types[field_name] = pd.Int64Dtype()
+            elif field_type in ["basic:decimal:"]:
+                column_types[field_name] = float
+
+        df = df.astype(column_types)
+
+        drop_columns = set(df.columns) - set(column_types.keys())
+        if drop_columns:
+            df = df.drop(columns=drop_columns)
+
+        return df
+
+    def _get_relations(self) -> pd.DataFrame:
+        relations = pd.DataFrame(index=[s.name for s in self._samples])
+        relations.index.name = "sample_name"
+
+        for relation in self.resolwe.relation.filter(collection=self.collection.id):
+            relations[relation.category] = pd.Series(
+                index=relations.index, dtype="object"
+            )
+
+            for partition in relation.partitions:
+                value = ""
+                if partition["label"] and partition["position"]:
+                    value = f'{partition["label"]} / {partition["position"]}'
+                elif partition["label"]:
+                    value = partition["label"]
+                elif partition["position"]:
+                    value = partition["position"]
+
+                sample_name = next(
+                    s.name for s in self._samples if s.id == partition["entity"]
+                )
+                relations[relation.category][sample_name] = value
+
+        return relations
+
+    def _get_orange_object(self) -> Data:
+        return self.collection.data.get(
+            process__slug="upload-orange-metadata",
+            ordering="-modified",
+            fields=DATA_FIELDS,
+            limit=1,
+        )
+
+    def _get_orange_data(self) -> pd.DataFrame:
+        try:
+            orange_meta = self._get_orange_object()
+        except LookupError:
+            return pd.DataFrame()
+
+        file_name = orange_meta.files(field_name="table")[0]
+        url = urljoin(self.resolwe.url, f"data/{orange_meta.id}/{file_name}")
+        response = self.resolwe.session.get(url, auth=self.resolwe.auth)
+        response.raise_for_status()
+
+        with BytesIO() as f:
+            f.write(response.content)
+            f.seek(0)
+            if any(file_name.endswith(ext) for ext in ["xls", "xlsx"]):
```

Comment:
Reading xls/xlsx files isn't that easy nowadays. [`xlrd`](https://pypi.org/project/xlrd/) doesn't support xlsx files anymore and `openpyxl` doesn't support xls. To support all excel files, we also need to add `xlrd` to dependencies and raise the minimum pandas version to 1.2.0 because `read_excel` just learned to handle it (see the beginning of [1.2.0 release notes](https://pandas.pydata.org/docs/whatsnew/v1.2.0.html)).

Reply:
OK, i just had an use case for `xlsx`, and enabled that... Thanks!